content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -722,6 +722,92 @@ controller is tracking a Job using Pod finalizers by checking if the Job has the
 annotation `batch.kubernetes.io/job-tracking`. You should **not** manually add
 or remove this annotation from Jobs.
 
+### Specifying your Pod failure policy
+
+{{< feature-state for_k8s_version="v1.25" state="alpha" >}}
+
+In some situations you may want to have a better control over handling of Pod
+failures than the default policy based on the Job's `.spec.backoffLimit` gives
+(see: [Pod backoff failure policy](#pod-backoff-failure-policy)).
+
+For example, you may want to terminate a job as soon as one of its Pods fails
+with an exit code indicating a software bug (for example due to an unknown
+command invoked). This may allow you to cut down costs of running computational
+workloads by avoiding unnecessary Pod restarts. Additionally, you may want to do
+not count pod failures caused by disruptions (such as preemption, eviction API
+or taint-based Pod deletions) towards the `.spec.backoffLimit` counter.
+
+You can satisfy the above use cases by specifying a Pod failure policy
+(in the `.spec.podFailrePolicy` field) for handling Pod failures based on the
+container exit codes and the Pod conditions.
+
+{{< note >}}
+You can only configure a Pod failure policy for a Job if you have the
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+enabled in your cluster: `JobPodFailurePolicy`. Additionally, it is recommended
+to enable the `PodDisruptionsCondition` in order to be able to detect and handle
+pod disruption conditions in the Pod failure policy. Both feature gates are
+available in Kubernetes v1.25.
+{{< /note >}}
+
+When you enable `PodDisruptionsCondition` in your cluster, a dedicated Pod
+`DisruptionTarget` condition is added to indicate a disruption. Its `reason`
+field additionally indicates one of the following reasons for the Pod
+termination:
+- `PreemptionByKubeScheduler` (Pod preempted by kube-scheduler)
+- `DeletionByTaintManager` (Pod deleted by taint manager due to NoExecute taint)
+- `EvictionByEvictionAPI` (Pod evicted by Eviction API)
+- `DeletionByPodGC` (an orphaned Pod deleted by PodGC)
+
+Example job configuration using this feature:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  template:
+    spec:
+      containers:
+      - name: example
+        image: example-job-image:v1
+        ...
+  backoffLimit: 6
+  podFailurePolicy:
+    rules:
+    - action: Ignore             # one of: Ignore, FailJob, Count
+      onPodConditions:
+      - type: DisruptionTarget   # indicates Pod disruption
+    - action: FailJob
+      onExitCodes:
+        containerName: example   # optional
+        operator: In             # one of: In, NotIn
+        values: [42]
+```
+
+The example job configuration exclude Pod disruptions from counting towards the
+`.spec.backoffLimit` counter. A container which exited with code 42 triggers
+termination of the job.
```

Comment:
Is this right?
```suggestion
`.spec.backoffLimit` counter. For the `example` container specifically:

- an exit status of 0 means that the container succeeded
- an exit status of 42 means that the **entire Job** has failed
- any other exit status represents a fault that is specific to the `example` container.

Because the Pod template specifies a `restartPolicy: Never`, the kubelet does
not restart the `example` container in that particular Pod for any exit code of `example`.
For example, if the `example` container exits with status 1, only the Pod where this failure
happens is affected.

If you trigger Job failure for a Job that is running multiple Pods, Kubernetes shuts down
all the Pods in that Job.
```